lib/validates_overlap/overlap_validator.rb

```diff
@@ -98,11 +98,14 @@ def generate_overlap_sql_conditions(record)
     if record.new_record?
       self.sql_conditions = main_condition
     else
-      self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key_name} != #{record.send(primary_key_name)}"
+      if record.send(primary_key_name).is_a?String
+        "#{main_condition} AND #{record_table_name(record)}.#{primary_key(record)} != '#{primary_key_value(record)}'"
+      else
+        "#{main_condition} AND #{record_table_name(record)}.#{primary_key(record)} != #{primary_key_value(record)}"
+      end
```

Comment:
Would break this out separately:

```ruby
else
  primary_key = record.send(primary_key_name)

  if primary_key.is_a?(String)
```

Also I think you may have an extra `)}` on the end of that first branch of the conditional.